zndraw/server/events.py

```diff
@@ -340,21 +359,38 @@ def scene_pause(data):
 @io.on("modifier:register")
 def modifier_register(data):
     data["token"] = session["token"]
+    if "modifier" not in app.config["PER-TOKEN-DATA"][session["token"]]:
+        # this is a dict for the pyclient, but it has to be for every webclient ...
+        app.config["PER-TOKEN-DATA"][session["token"]]["modifier"] = {}
 
     try:
         # we can only register one modifier at a time
         name = data["modifiers"][0]["name"]
-        if name in app.config["MODIFIER"]:
-            # issue with the same modifier name on different webclients / tokens!
-            # only for default we need to ensure, there is only one.
-            raise ValueError(f"Modifier {name} is already registered.")
-        app.config["MODIFIER"][name] = request.sid
+        # get the key from the value request.sid by inverting the dict
+        _tmp = {v: k for k, v in app.config["pyclients"].items()}
+        app.config["PER-TOKEN-DATA"][session["token"]]["modifier"][name] = _tmp[
```

Comment:
raise an error if the modifier is already registered. You don't want to run register again, but instead it should be handled by updating the SID-Token map